crates/provider-messaging-kafka/src/lib.rs

```diff
@@ -298,20 +323,35 @@ impl exports::wasmcloud::messaging::consumer::Handler<Option<Context>> for Kafka
             .produce(records, Compression::default())
             .await
         {
-            error!("failed to produce record: {e}");
+            error!(error = %e, "failed to produce record");
             return Ok(Err(format!("failed to produce record: {e}")));
         }
         Ok(Ok(()))
     }
 
-    #[instrument(level = "debug", skip_all, fields(subject = _subject))]
+    #[instrument(skip_all)]
     async fn request(
         &self,
-        _ctx: Option<Context>,
-        _subject: String,
-        _body: Vec<u8>,
-        _timeout_ms: u32,
+        ctx: Option<Context>,
+        subject: String,
+        body: Vec<u8>,
+        timeout_ms: u32,
     ) -> anyhow::Result<Result<BrokerMessage, String>> {
+        // Extract tracing information from invocation context, if present
+        let trace_ctx = match ctx {
+            Some(Context { ref tracing, .. }) if !tracing.is_empty() => tracing
+                .iter()
+                .map(|(k, v)| (k.to_string(), v.to_string()))
+                .collect::<Vec<(String, String)>>(),
+
+            _ => TraceContextInjector::default_with_span()
+                .iter()
+                .map(|(k, v)| (k.to_string(), v.to_string()))
+                .collect(),
+        };
+        wasmcloud_tracing::context::attach_span_context(&trace_ctx);
+        debug!(?subject, ?body, ?timeout_ms, "performing request");
```

Comment:
I think this statement is confusing, since nothing is actually being done

Reply:
Sure -- will remove!

- [ ] remove performing request debug